### ISSUE TITLE
As a user, I want to see the last time a remote was synced.

### ISSUE THREAD

**Issue by newswangerd** [pulp/pulpcore#2863]
**Is your feature request related to a problem? Please describe.**
When I look through the remotes on the API, I want to know the details of the last time the remote was synced. This information helps me understand how up to date my content is and alert me if any syncs are failing.

**Describe the solution you'd like**
On the remote API response I would like to see a field like this:

```json
            "last_sync_task": {
                "pulp_href": "/pulp/api/v3/tasks/abf24c48-cc91-4ac2-b628-5b05f06c218f/",
                "state": "running"
            },
```

This can be accomplished by looking up the latest task that includes the given remote in `reserved_resource_record`. There are two options for serializing this. The task can either be added directly to the serializer like the example above, or returned as an href like so `"last_sync_href": "/pulp/api/v3/tasks/abf24c48-cc91-4ac2-b628-5b05f06c218f/"` with the option to view it in more detail through a feature such as related fields (https://github.com/pulp/pulpcore/pull/2828).

**Additional context**
This is a nice to have performance improvement for https://issues.redhat.com/browse/AAH-756. Without this feature, the galaxy ui will have to query `/pulp/api/v3/tasks/?reserved_resource_record` every time it needs to display sync information for a remote in the UI.

**Comment by newswangerd** [comment]
@ipanova would it make sense to put something like this on the repository api instead? If syncs take exclusive access to the repo, then you should only be able to do one sync at a time, and I think it would make sense to show the latest, or currently running sync on a repo, even if the repo can be synced via multiple remotes.

Also, there's a remote field on the repository model (https://github.com/pulp/pulpcore/blob/main/pulpcore/app/models/repository.py#L67). If the apis allow you to sync any remote with any repo, what is this used for?

This is definitely a nice to have. It lets us create nice UIs like this one where the user can see the status of all their remote syncs in one place, but it might make sense to just redesign the UI.
![Screen Shot 2022-06-16 at 12 23 06 PM](https://user-images.githubusercontent.com/6063371/174119042-058006a0-2890-4ea3-b5a8-5bdc29402440.png)

**Comment by mdellweg** [comment]
I would say the repository version could use some information of how it got created, like a link to the task & the used remote. Both with on_delete=set_null to avoid strong object dependencies. The information of when the version was created is already there.

**Comment by ipanova** [comment]
> Also, there's a remote field on the repository model (https://github.com/pulp/pulpcore/blob/main/pulpcore/app/models/repository.py#L67). If the apis allow you to sync any remote with any repo, what is this used for?

When creating the repo you can assign a remote to it. So whenever you trigger sync on that repo you don't need to provide the repo in the api call every time. However if you do provide, than the provided one will be used instead of the 'default' one .

**Comment by ipanova** [comment]
@newswangerd in the meantime we have added this extra information to the repo_version https://github.com/pulp/pulpcore/issues/2998 It is not serialized, so not user visible, but the information can still be extracted. Is this something that can be used to address this request?

**Comment by newswangerd** [comment]
@ipanova it looks like the info field is blank, right? Is the idea here that we can add information about the last sync to the info field if we want?

**Comment by ipanova** [comment]
> @ipanova it looks like the info field is blank, right? Is the idea here that we can add information about the last sync to the info field if we want?

yes, the idea is to store whatever extra information is needed on the repo_version.
